lib/bootloader_setup.pm

```diff
@@ -214,9 +214,15 @@ sub specific_bootmenu_params {
         diag "Disabling installer self update as requested by INSTALLER_NO_SELF_UPDATE=1";
         $args .= " self_update=0";
     }
-    elsif (check_var("INSTALLER_SELF_UPDATE", 1)) {
-        diag "Explicitly enabling installer self update as requested by INSTALLER_SELF_UPDATE=1";
-        $args .= " self_update=1";
+    elsif (my $self_update_repo = get_var("INSTALLER_SELF_UPDATE")) {
+        if ($self_update_repo eq "1") {
+            diag "Explicitly enabling installer self update as requested by INSTALLER_SELF_UPDATE=1";
```

Comment:
do we really need this to be so verbose? I would just take the argument and add diag some generic text (if it at all)